### PR TITLE
Allow to add url without TLD to install_app

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...I want to allow install app with url without TLD

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-2236 `install_app` rejects valid URLs](https://app.clickup.com/t/2549495/SALEOR-2236)

File: saleor/core/utils/url.py [hunk 1]
```diff
@@ -33,3 +25,16 @@ def prepare_url(params: str, redirect_url: str) -> str:
...
+    try:
+        parsed_url = urlparse(url)
+        domain, _ = split_domain_port(parsed_url.netloc)
+        if not parsed_url.netloc:
+            raise ValidationError(
+                "Invalid URL. Please check if URL is in RFC 1808 format."
+            )
+    except ValueError as error:
+        raise ValidationError(error)
+    return domain
```
Comment (IKarbowiak) [inline saleor/core/utils/url.py:40]: Do we have tests for this?

**Review by IKarbowiak — APPROVED** [review]

**Comment by d-wysocki** [comment]
Should we validate if this url it's in `ALLOWED_HOSTS`?

File: saleor/core/utils/url.py (hunk 1 above)
Comment (korycins) [inline saleor/core/utils/url.py:40]: Personally, it is strange that a function with the name  `validate_url` returns the domain.  I wouldn't expect that the validation will return me a part of url.

Comment (korycins) [reply saleor/core/utils/url.py:40]: I don't think that this method does the same job as URLValidation. I understand that the issue was TLD, but why we didn't extend the URLValidator to handle the expected cases? Now, from a fully validated URL, we moved to parse URL. 
I would start with something like this:
```
valdiator = URLValidator()
validator.tld_re = "your regex"
valdiator.host_re = '(' + hostname_re + domain_re + tld_re + '|localhost)'
```

Comment (d-wysocki) [reply saleor/core/utils/url.py:40]: Good point, I will change that.

File: saleor/core/utils/url.py [hunk 2]
```diff
@@ -33,3 +25,16 @@ def prepare_url(params: str, redirect_url: str) -> str:
     split_url = urlsplit(redirect_url)
     split_url = split_url._replace(query=params)
     return split_url.geturl()
+
+
+def validate_url(url):
```
Comment (fowczarek) [inline saleor/core/utils/url.py:30]: We should add docstring to this function.

**Review by fowczarek — APPROVED** [review]

**Comment by korycins** [comment]
@d-wysocki  also one important thing. The App model uses URLField which uses a standard URLValidator for checking the value. We need to take this into account during applying this changes, afaik.

File: saleor/core/utils/url.py [hunk 3]
```diff
@@ -33,3 +36,17 @@ def prepare_url(params: str, redirect_url: str) -> str:
     split_url = urlsplit(redirect_url)
     split_url = split_url._replace(query=params)
     return split_url.geturl()
+
+
+class APPURLValidator(URLValidator):
+    validator = URLValidator
+    host_re = "(" + validator.hostname_re + validator.domain_re + "|localhost)"
+    regex = _lazy_re_compile(
+        r"^(?:[a-z0-9.+-]*)://"  # scheme is validated separately
```
Comment (maarcingebala) [inline saleor/core/utils/url.py:45]: Typo?
```suggestion
        r"^(?:[a-z0-9.+-]*)://"  # schema is validated separately
```

File: saleor/core/utils/url.py [hunk 4]
```diff
@@ -33,3 +36,17 @@ def prepare_url(params: str, redirect_url: str) -> str:
     split_url = urlsplit(redirect_url)
     split_url = split_url._replace(query=params)
     return split_url.geturl()
+
+
+class APPURLValidator(URLValidator):
```
Comment (maarcingebala) [inline saleor/core/utils/url.py:41]: I think it should say: `AppURLValidator`

Comment (maarcingebala) [reply saleor/core/utils/url.py:41]: I also think this validator could be part of the `saleor/app` module. It's specific to apps; also it is used only in app-related modules.

File: saleor/core/utils/url.py (hunk 3 above)
Comment (d-wysocki) [reply saleor/core/utils/url.py:45]: `Scheme` its for URLs
https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_is_a_URL#scheme

Comment (maarcingebala) [reply saleor/core/utils/url.py]: 👍

File: saleor/app/validators.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+from django.core.validators import URLValidator
+from django.utils.regex_helper import _lazy_re_compile  # type: ignore
+
+
+class AppURLValidator(URLValidator):
+    validator = URLValidator
+    host_re = "(" + validator.hostname_re + validator.domain_re + "|localhost)"
+    regex = _lazy_re_compile(
+        r"^(?:[a-z0-9.+-]*)://"  # scheme is validated separately
```
Comment (fowczarek) [inline saleor/app/validators.py:11]: Could we add the reason why the `scheme is validated separately` in this comment?

Comment (d-wysocki) [reply saleor/app/validators.py:11]: It's took directly from django, that will be okay if I will add source of this piece of code?

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**